src/axom/sidre/core/MFEMSidreDataCollection.hpp

```diff
@@ -299,6 +304,57 @@ class MFEMSidreDataCollection : public mfem::DataCollection
       deleting it from the named_buffers group, if allocated. */
   virtual void DeregisterField(const std::string& field_name);
 
+  /// Associates a field name with a multi-buffer material set
+  /** Subsequent calls to RegisterField with field_names of the form
+   * @p volume_fraction_field_name_<material_id> will result in the addition
+   * of a volume buffer to the matset @p matset_name corresponding to @p material_id.
+   * 
+   * Note that this does not inhibit the addition of the field - that is, the GridFunction
+   * data will be present as both a field and a volume fraction buffer
+   * 
+   * @param volume_fraction_field_name The field name to associate with volume fractions
+   * @param matset_name The material set to associate subsequently added volume fractions with
+   *
+   */
+  void AssociateMaterialSet(const std::string& volume_fraction_field_name,
+                            const std::string& matset_name);
+
+  /// Associates a field name with a species set
+  /** Subsequent calls to RegisterField with field_names of the form
+   * @p species_field_name_<material_id>_<component> will result in the addition
+   * of its values to the specset @p specset_name corresponding to @p material_id.
+   * and specified component
+   * 
+   * Note that this does not inhibit the addition of the field - that is, the GridFunction
+   * data will be present as both a field and within the species set
+   * 
+   * @param species_field_name The field name to associate with the species set matset values
+   * @param specset_name The name of the species set to associate added matset values with
+   * @param matset_name The material set to associate with the species set
+   * @param volume_dependent Whether the species set is volume-dependent
+   *
+   */
+  void AssociateSpeciesSet(const std::string& species_field_name,
+                           const std::string& specset_name,
+                           const std::string& matset_name,
+                           const bool volume_dependent);
+
+  /// Associates a material-dependent field with its corresponding material set
+  /** Subsequent calls to RegisterField with field_names of the form
+   * @p dependent_field_name_<material_id> will result in the addition
+   * of its values to the to the matset_vals of the @p dependent_field_name corresponding to
+   * @p material_id
+   * 
+   * Note that this does not inhibit the addition of the field - that is, the GridFunction
+   * data will be present as both a field on its own and as part of the "top-level" field
+   * 
+   * @param material_dependent_field_name The name of the field to mark as material-dependent
+   * @param matset_name The material set to associate with the field
+   *
```

Comment:
Is it an error to call this if  `matset_name` was not registered first via `AssociateMaterialSet()` ?